workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -810,5 +810,45 @@ for (var j = 0; j < commandDetailsList.Count; j++) {
                 chunkArray.Dispose();
             }
         }
+
+        internal class AcknowledgeAuthorityLossHandler : AbstractAcknowledgeAuthorityLossHandler
+        {
+            public override EntityArchetypeQuery Query => new EntityArchetypeQuery
+            {
+                All = new ComponentType[]
+                {
+                    ComponentType.ReadOnly<AuthorityLossImminent<<#= componentNamespace #>.Component>>(),
+                    ComponentType.ReadOnly<SpatialEntityId>()
+                },
+                Any = Array.Empty<ComponentType>(),
+                None = Array.Empty<ComponentType>()
+            };
+
+            public override void AcknowledgeAuthorityLoss(ComponentGroup group, ComponentSystemBase system,
+                Improbable.Worker.Core.Connection connection)
+            {
+                var authorityLossType = system.GetArchetypeChunkComponentType<AuthorityLossImminent<<#= componentNamespace #>.Component>>();
+                var spatialEntityType = system.GetArchetypeChunkComponentType<SpatialEntityId>();
+
+                var chunkArray = group.CreateArchetypeChunkArray(Allocator.TempJob);
+
+                foreach (var chunk in chunkArray)
+                {
+                    var authorityArray = chunk.GetNativeArray(authorityLossType);
+                    var spatialEntityIdArray = chunk.GetNativeArray(spatialEntityType);
+
+                    for (int i = 0; i < authorityArray.Length; ++i)
+                    {
+                        if (authorityArray[i].AcknowledgeAuthorityLoss)
+                        {
+                            connection.SendAuthorityLossImminentAcknowledgement(spatialEntityIdArray[i].EntityId,
+                                <#= componentDetails.ComponentId #>);
+                        }
```

Comment:
Would the `AcknowledgeAuthorityLoss` property need to be reset to false here? Or is the owning struct cleaned up in this frame?

Reply:
No it shouldn't be reset. If it's set to true then it will be cleaned up by the component being removed.

Reply:
When does the component get removed? If it gets removed in this frame or the beginning of next frame all good, but is there a chance of multiple frames / ticks having this component with `AcknowledgeAuthorityLoss=true`?

Reply:
That's intended. You can't acknowledge authority loss multiple times. It's a flag which you set once.